src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -374,7 +374,7 @@ internal static unsafe void DecimalToNumber(ref decimal d, ref NumberBuffer numb
             }
             *dst = (byte)('\0');
 
-            number.CheckConsistency();
+            number.CheckConsistency(skipTrailingZeroCheck: true);
```

Comment:
This is because `decimal` needs to care about the number of trailing zeros, correct?

Reply:
Yup, correct.

Reply:
I meant that for decimal, `1000.0` and `1000.00` are represented differently.

```
> decimal.GetBits(1000.0m)
int[4] { 10000, 0, 0, 65536 }
> decimal.GetBits(1000.00m)
int[4] { 100000, 0, 0, 131072 }
```

because the number of zeros are actually tracked